lib/article_rating_extractor.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+#= Fetches article classes from PageAssessments response
+# We simplify this parser by folding the nonstandard ratings
+# into the corresponding standard ones. We don't want to deal with edge cases
+# like bplus and a/ga.
+
+class ArticleRatingExtractor
+  def initialize(pageassessments)
+    @assessments = pageassessments || {}
+  end
+
+  ################
+  # Entry point #
+  ################
+
+  def ratings
+    # Returns the classes of articles mapped to the respective title
+    ratings = {}
+    @assessments.values.each do |page|
+    # The class of the first pageassessment is taken
+      rating =  ratings[page['title'].tr(' ', '_')]
+      ratings[page['title'].tr(' ', '_')] ||= get_article_class(page['pageassessments'])
+    end
+    return ratings
+  end
+
+  private
+
+  def get_article_class(pageassessments)
+    # Returns nil if no page assessment data found
+    return nil if pageassessments.nil?
+    return nil unless pageassessments.is_a? Enumerable
```

Comment:
What is the purpose of this check, beyond the nil check? Did you find cases where this would receive a non-enumerable non-nil argument, or is it just double-checking to making sure the following operation is safe?